src/Microsoft.AspNetCore.Routing/Constraints/RegexInlineRouteConstraint.cs

```diff
@@ -13,7 +13,7 @@ public class RegexInlineRouteConstraint : RegexRouteConstraint
         /// </summary>
         /// <param name="regexPattern">The regular expression pattern to match.</param>
         public RegexInlineRouteConstraint(string regexPattern)
-            : base(regexPattern)
+            : base("^" + regexPattern + "$")
```

Comment:
I was thinking about that as well, initially..

After your comment and while typing a reply I did several tests on both the old and the new framework, and realised that inline regex constraints work differently to the constraints route parameter. So while the constraints route parameter creates a bounded regex match, the inline constraints does not. If I remember correctly, the inline constraints were initially added in WebAPI 2, and the behaviour there was the same as it is currently in ASP.NET Core.

To summarise, given the following routes:

    routes.MapRoute(
        name: "test1",
        template: "test/{value}",
        defaults: new { controller = "Test", action = "Index" },
        constraints: new { value = "val.*" });

    routes.MapRoute(
        name: "test3",
        template: "test2/{value:regex(val.*)}",
        defaults: new { controller = "Test", action = "Index" });

We have the following behaviour:

 * `/test/val123' => Test.Index()
 * `/test/xval123' => 404
 * `/test2/val123' => Test.Index()
 * `/test2/xval123' => Test.Index()

While I find this behaviour is potentially a cause for confusion and source of potentially bad routes, I also understand that it's been like this since WebAPI 2, so it's definitely not an easy change to make.

With that in mind, I think I understand why my original request might not be such a great idea, but at the same time I'm not very pleased with the above either. It would be great if there was a way to somehow warn the users of this difference in behaviour.

I'm not 100% sure what would be the best decision here (perhaps a `FullMatchRegexInlineRouteConstraint` might be a good solution after all), but at the very least it shows that the current doc page is wrongly describing how the inline regex constraint works, so even if the above pull request is rejected, the docs have to be fixed.

I can update the pull request with the reversed behaviour,